src/modules/feedback/components/rating/FiveButtonRating.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @module modules/feedback/components/rating/FiveButtonRating
+ */
+
+import { html } from 'lit-html';
+import css from './fiveButtonRating.css';
+import { MvuElement } from '../../../MvuElement';
+import { $injector } from '../../../../injection';
+
+/**
+ * possible rating types
+ * @readonly
+ * @enum {string}
+ */
+export const Rating = Object.freeze({
+	NONE: '0',
+	TERRIBLE: '1',
+	BAD: '2',
+	SATISFIED: '3',
+	GOOD: '4',
+	EXCELLENT: '5'
+});
+
+const Update_Rating = 'update_rating';
+
+export class FiveButtonRating extends MvuElement {
+	constructor() {
+		super({
+			rating: Rating.NONE
+		});
+
+		const { TranslationService: translationService } = $injector.inject('TranslationService');
+		this._translationService = translationService;
+	}
+
+	update(type, data, model) {
+		switch (type) {
+			case Update_Rating:
+				return { ...model, rating: data };
+		}
+	}
+
+	createView(model) {
+		const { rating } = model;
+
+		const translate = (key) => this._translationService.translate(key);
+
+		return html`
+			<style>
+				${css}
+			</style>
+
+			<div>
+				<button
+					class="star-button ${Rating.TERRIBLE <= rating ? 'selected' : 'unselected'}"
+					@click="${() => this._onRatingClick(Rating.TERRIBLE)}"
+					title="${translate('fiveButtonRating_terrible')}"
+				>
+					*
+				</button>
+				<button
+					class="star-button ${Rating.BAD <= rating ? 'selected' : 'unselected'}"
+					@click="${() => this._onRatingClick(Rating.BAD)}"
+					title="${translate('fiveButtonRating_bad')}"
+				>
+					*
+				</button>
+				<button
+					class="star-button ${Rating.SATISFIED <= rating ? 'selected' : 'unselected'}"
+					@click="${() => this._onRatingClick(Rating.SATISFIED)}"
+					title="${translate('fiveButtonRating_satisfied')}"
+				>
+					*
+				</button>
+				<button
+					class="star-button ${Rating.GOOD <= rating ? 'selected' : 'unselected'}"
+					@click="${() => this._onRatingClick(Rating.GOOD)}"
+					title="${translate('fiveButtonRating_good')}"
+				>
+					*
+				</button>
+				<button
+					class="star-button ${Rating.EXCELLENT <= rating ? 'selected' : 'unselected'}"
+					@click="${() => this._onRatingClick(Rating.EXCELLENT)}"
+					title="${translate('fiveButtonRating_excellent')}"
+				>
+					*
+				</button>
+			</div>
+		`;
+	}
+
+	_onRatingClick(rating) {
+		this.rating = rating;
+	}
+
+	set rating(value) {
+		this.signal(Update_Rating, value);
+		this.dispatchEvent(
```

Comment:
Here we would fire the event every time the setter is called even if the model hasn't changed.
We could improve this by registering an observer for our model (in onInitialize()) and firing the event when our observer was called.